application/event_sender/tests/test_event_sender.py

```diff
@@ -135,7 +139,7 @@ def test_lambda_handler_dos_api_fail(
     mock_put_metric.assert_has_calls([call("DosApiLatency", 0, "Milliseconds"), call("DoSApiFail", 1, "Count")])
     assert response["statusCode"] == 500
     assert response["body"] == "something went wrong"
-    mock_client.return_value.delete_message.asset_not_called()
```

Comment:
good spot, this is a bit of an issue, with no type checking its easy to miss this as the test would have passed even if the function had been called